lib/News.ts

```diff
@@ -22,38 +25,26 @@ export interface NewsItem extends NewsCard {
     content: string;
 }
 
-const tempNewsItems = [
-    {
-        id: 0,
-        date: 1693961460,
-        title: "The Club Website is Up and Running!",
-        description:
-            "The CS Club has a brand new website so people can learn about the club and events without needing a Discord account.",
-        author: "Jonah Uellenberg",
-        content: `# Club Website
-The club website is still a work-in-progress, but it's officially complete enough to serve as the club's front page! Currently, the website supports
-giving people information about the club, directing them towards joining the Discord server, and showing recent events/news.
-
-## Contributing
-There's still a lot that can be added to the website. If you're interested in learning about web development and working on the website with other students, then
-stay tuned for a Club Website event. Alternatively, you can view and contribute to the website's source code at https://github.com/nsccs/club_website.`,
-    },
-];
-
-/**
- * Gets a news item, or null if it doesn't exist.
- * @param id - is the ID to look up the news item.
- */
-export async function getNewsItem(id: number): Promise<NewsItem | null> {
-    // TODO: Implement.
-    return tempNewsItems[id] || null;
-}
-
 /**
  * Gets the data needed to display several news cards.
  * @param count - is the maximum number of entries to return.
  */
-export async function getNewsCards(count: number): Promise<NewsCard[]> {
-    // TODO: Implement.
-    return tempNewsItems.slice(0, count);
+export async function getNewsCards(count: number): Promise<NewsItem[] | null> {
+    // Try to use cached data.
+    const cached = await redisClient.get("cache_news_cards:" + count);
+    if (cached) {
+        return JSON.parse(cached);
+    }
+    const news = await getNews(count);
+    news.map(row => {
+        row.date = row.date.toString();
+        return row;
+    });
+    // Save the data in the cache.
+    await redisClient.setex(
+      "cache_news_cards:" + count,
+      DAY,
```

Comment:
I would reduce this time a bit (and it probably should be reduced with events as well). Events are sort of expensive to request because they involve a lot of HTTP requests to an external server, but all the DB stuff is internal and can be requested a bit more frequently. Maybe set it to an HOUR or so, but that's up to you.